docs/model-dev-guide/batch-processing/batch-process-api-ug.rst

```diff
@@ -0,0 +1,173 @@
+runs distributed data processing.
+
+This API automatically handles the following for you:
+
+-  shard a dataset by number of workers available
+-  apply user-defined logic to each batch of data
+-  handle synchronization between workers
+-  track job progress to enable preemption and resumption of trial
+
+This is a flexible API that can be used for many different tasks, including batch (offline)
+inference.
+
+*******
+ Usage
+*******
+
+The main arguments to :meth:`~determined.pytorch.experimental.torch_batch_process` are processor
+class and dataset.
+
+.. code:: python
+
+   torch_batch_process(
+       batch_processor_cls=MyProcessor
+       dataset=dataset
+   )
+
+``TorchBatchProcessor``
+=======================
+
+During :meth:`~determined.pytorch.experimental.TorchBatchProcessor.__init__` of
+:class:`~determined.pytorch.experimental.TorchBatchProcessor`, we pass in a
+:class:`~determined.pytorch.experimental.TorchBatchProcessorContext` object, which contains useful
+methods that can be used within the :class:`~determined.pytorch.experimental.TorchBatchProcessor`
+class.
+
+:class:`~determined.pytorch.experimental.TorchBatchProcessor` is compatible with Determined's
+:class:`~determined.pytorch.MetricReducer`. You can pass MetricReducer to
+:class:`~determined.pytorch.experimental.TorchBatchProcessor` as follow:
+
+``TorchBatchProcessorContext``
+==============================
+
+:class:`~determined.pytorch.experimental.TorchBatchProcessorContext` should be a subclass of
+:class:`~determined.pytorch.experimental.TorchBatchProcessor`. The two functions you must implement
+are the :meth:`~determined.pytorch.experimental.TorchBatchProcessor.__init__` and
+:meth:`~determined.pytorch.experimental.TorchBatchProcessor.process_batch`. The other lifecycle
+functions are optional.
+
+.. code:: python
+
+   class MyProcessor(TorchBatchProcessor):
+       def __init__(self, context):
+           self.reducer = context.wrap_reducer(reducer=AccuracyMetricReducer(), name="accuracy")
+
+******************************************
+ How To Perform Batch (Offline) Inference
+******************************************
+
+In this section, we'll learn how to perform batch inference using the Torch Batch Processing API.
+
+Step 1: Define a InferenceProcessor
+===================================
+
+The first step is to define an InferenceProcessor. You should initialize your model in the
+:meth:`~determined.pytorch.experimental.TorchBatchProcessor.__init__` function of the
+InferenceProcessor. You should implement
+:meth:`~determined.pytorch.experimental.TorchBatchProcessor.process_batch` function with inference
+logic.
+
+You can optionally implement
+:meth:`~determined.pytorch.experimental.TorchBatchProcessor.on_checkpoint_start` and
+:meth:`~determined.pytorch.experimental.TorchBatchProcessor.on_finish` function to be run before
+every checkpoint and after all the data are processed respectively.
+
```

Comment:
You can optionally implement
:meth:`~determined.pytorch.experimental.TorchBatchProcessor.on_checkpoint_start` and
:meth:`~determined.pytorch.experimental.TorchBatchProcessor.on_finish` to be run before
every checkpoint and after all the data has been processed, respectively.